astropy/io/fits/diff.py

```diff
@@ -273,7 +273,7 @@ def __init__(self, a, b, ignore_hdus=[], ignore_keywords=[],
                 a = fitsopen(a)
             except Exception as exc:
                 raise OSError("error opening file a ({}): {}: {}".format(
-                        a, exc.__class__.__name__, exc.args[0]))
```

Comment:
again, maybe consider breaking with `.format()`. But then this is not a strong preference, and I won't comment any of the cases below.